spiped/main.c

```diff
@@ -340,6 +343,9 @@ main(int argc, char * argv[])
 		goto err6;
 	}
 
+	/* Deregister the graceful_shutdown timer. */
+	events_timer_cancel(graceful_shutdown_timer_cookie);
```

Comment:
... then don't we end up trying to cancel a timer which doesn't exist here?

I suggest setting to NULL when the callback is invoked and then checking for NULL here.